unity-client/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/WebRequestBase.cs

```diff
@@ -0,0 +1,96 @@
+using DCL.Helpers;
+using System;
+using System.Collections;
+using System.Net.Http;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace DCL
+{
+    /// <summary>
+    /// Base class for all our custom WebRequest implementations that will manage the core lyfecicle of the requests, including the retry system.
+    /// </summary>
+    public interface IWebRequestBase
+    {
+        /// <summary>
+        /// Request and download data from a url (synchronous mode).
+        /// </summary>
+        /// <param name="url">Url where to make the request.</param>
+        /// <param name="requestAttemps">Number of attemps for re-trying failed requests.</param>
+        /// <returns>The web request with the data downloaded.</returns>
+        UnityWebRequest Get(string url, int requestAttemps = 3);
+
+        /// <summary>
+        /// Download data from a url (asynchronous mode).
+        /// </summary>
+        /// <param name="url">Url where to make the request.</param>
+        /// <param name="OnCompleted">This action will be executed if the request successfully finishes and it includes the request with the data downloaded.</param>
+        /// <param name="OnFail">This action will be executed if the request fails.</param>
+        /// <param name="requestAttemps">Number of attemps for re-trying failed requests.</param>
+        UnityWebRequest GetAsync(string url, Action<UnityWebRequest> OnCompleted, Action<string> OnFail, int requestAttemps = 3);
+    }
+
+    public abstract class WebRequestBase : IWebRequestBase
+    {
+        public UnityWebRequest Get(string url, int requestAttemps = 3)
+        {
+            UnityWebRequest request;
+            int remainingAttemps = Mathf.Clamp(requestAttemps, 1, requestAttemps);
+
+            do
+            {
+                try
+                {
+                    request = CreateWebRequest(url);
+                    var requestOperation = request.SendWebRequest();
+                    while (!requestOperation.isDone && requestOperation.progress < 1) { }
```

Comment:
```suggestion
                    while (!requestOperation.isDone && requestOperation.progress < 1) { }
```

This is OK for editor mode requests, but should never happen in runtime. We should always make the requests async by default. I suggest we remove this `Get` method altogether and rename the `GetAsync` to `Get` to avoid this case entirely.

Reply:
Yes, I implemented the sync GET method thinking in it could be useful in some situation, but you're right, we are not going to use it in any context. I have removed it and renamed the `GetAsync` by `Get`. Thanks Brian.